test/functional/vimscript/screenchar_spec.lua

```diff
@@ -79,4 +79,34 @@ describe('screenchar() and family respect floating windows', function()
   describe('without ext_multigrid', function()
     with_ext_multigrid(false)
   end)
+
+  describe('hidden windows', function()
+    before_each(function()
+      clear()
+      Screen.new(40, 7, {})
+      api.nvim_buf_set_lines(0, 0, -1, true, { 'aaa', 'aaa' })
+    end)
+
+    local validate = function()
```

Comment:
let's call this `assert_something`, since that's the common convention here, and `validate` is very well established for `vim.validate`. also we always use `local function` where possible.
```suggestion
    local function assert_...()
```